daemon/api_snaps.go

```diff
@@ -464,19 +467,67 @@ func snapInstall(ctx context.Context, inst *snapInstruction, st *state.State) (s
 		ckey = strutil.ElliptLeft(inst.CohortKey, 10)
 		logger.Noticef("Installing snap %q from cohort %q", inst.Snaps[0], ckey)
 	}
-	tset, err := snapstateInstall(ctx, st, inst.Snaps[0], inst.revnoOpts(), inst.userID, flags)
+
+	// TODO:COMPS: handle installing a component of a snap that is already
+	// installed
+	goal := storeInstallGoalFromInstruction(inst)
+	_, tset, err := snapstateInstallOne(ctx, st, goal, snapstate.Options{
+		UserID: inst.userID,
+		Flags:  flags,
+	})
 	if err != nil {
 		return "", nil, err
 	}
 
-	msg := fmt.Sprintf(i18n.G("Install %q snap"), inst.Snaps[0])
+	return installMessage(inst, ckey), []*state.TaskSet{tset}, nil
+}
+
+func installMessage(inst *snapInstruction, cohort string) string {
+	var b strings.Builder
+	fmt.Fprintf(&b, i18n.G("Install %q snap"), inst.Snaps[0])
```

Comment:
it's a bit weird that this is translatable

Reply:
I tried to rework this in fce48da2ee6dc57ae8d097b9b622df3691fbead8, but I'm not so sure that it is done correctly.